src/__tests__/components/fields/text-field/text-field.spec.tsx

```diff
@@ -105,9 +107,60 @@ describe(DEFAULT_FIELD_TYPE, () => {
 		expect(getTextfield()).toHaveAttribute("readonly");
 		expect(getTextfield()).toBeDisabled();
 	});
+
+	it("should not prevent copy and paste", async () => {
+		renderComponent({
+			customOptions: {
+				preventCopyAndPaste: false,
+			},
+		});
+		const textField = getTextfield();
+		textField.focus();
+		await act(async () => {
+			await userEvent.paste(EXPECTED_TEXT);
+			expect(textField).toHaveValue(EXPECTED_TEXT);
+		});
+	});
+
+	it("should prevent copy and paste", async () => {
+		renderComponent({
+			customOptions: {
+				preventCopyAndPaste: true,
+			},
+		});
+		const textField = getTextfield();
+		textField.focus();
+		await act(async () => {
+			await userEvent.paste(EXPECTED_TEXT);
+			expect(textField).not.toHaveValue(EXPECTED_TEXT);
+		});
+	});
+
+	it("should fireEvent return false since default behavior is prevented", () => {
+		renderComponent({
+			customOptions: {
+				preventDragAndDrop: true,
+			},
+		});
+		const textField = getTextfield();
+		const event = fireEvent.drop(textField, { target: { value: EXPECTED_TEXT } });
+		expect(event).toBe(false);
+	});
+
+	it("should fireEvent return true since default behavior is not prevented", () => {
```

Comment:
- `should allow drag & drop text into field if preventDragAndDrop is false`
- add a comment before this test to explain why the test is written this way
`// testing the return value of drop event because @testing-library/user-event does not respect preventDefault()`
- add `expect(textField).toHaveValue(EXPECTED_TEXT)`